### PR TITLE
Improve test platform

### PR DESCRIPTION
**Pull request by henrykironde**

### CODE REVIEW

File: tests/testthat/test-retriever.R [hunk 1]
```diff
@@ -1,7 +1,107 @@
-library(testthat)
-library(rdataretriever)
+context("regresion tests")
```
Comment (ethanwhite) [inline tests/testthat/test-retriever.R]: "regresion" -> "regression"

File: tests/testthat/test-retriever.R [hunk 2]
```diff
@@ -1,7 +1,107 @@
...
 })
+
+
+test_that("Download the raw portal dataset into './data/'", {
+  portal <- list("3299474", "3299483", "5603981")
+  rdataretriever::download('portal', './data/')
+  for(file in portal)
+  {
+    file_path <- normalizePath(file.path(getwd(), "tests/testthat/data/", file), winslash = "//", mustWork = FALSE)
+    expect_identical((file.info(file_paths)$size == 0)==FALSE, TRUE)
```
Comment (ethanwhite) [inline tests/testthat/test-retriever.R]: Do we need to do the `==FALSE` and test against `TRUE` or could we drop `==FALSE` and test against `FALSE`? 

E.g.,

```
expect_identical((file.info(file_paths)$size == 0), FALSE)
```

Or better yet, use the `expect_gt` function (which checks to see if the first value is greater than the second):

```
expect_gt(file.info(file_paths)$size, 0)
```


Same question applies several times below.

Comment (ethanwhite) [reply tests/testthat/test-retriever.R]: Where does `file_paths` come from? Is it supposed to be `file_path`? Same question applies below.

File: tests/testthat/test-retriever.R [hunk 3]
```diff
@@ -1,7 +1,107 @@
...
+})
+
+
+test_that("Install the portal into csv", {
+  # Install portal into csv files in your working directory
+  portal <- list("portal_main", "portal_plots", "portal_species")
+  rdataretriever::install('portal', 'csv')
+  for(file in portal)
+  {
+    file_path <- normalizePath(file.path(getwd(), "tests/testthat/", paste(file, ".csv")), winslash = "//", mustWork = FALSE)
```
Comment (ethanwhite) [inline tests/testthat/test-retriever.R]: Add newlines before `winslash` and `mustWork` to keep line length reasonable. (same below and probably above for consistency)

File: tests/testthat/test-retriever.R [hunk 4]
```diff
@@ -1,7 +1,107 @@
...
+
+
+test_that("Install the portal into sqlite", {
+  # Install the portal into Sqlite
+  portal <- list("portal_main", "portal_plots", "portal_species")
+  rdataretriever::install('portal', 'sqlite', db_file="Test.sqlite")
+  con <- dbConnect(RSQLite::SQLite(), dbname="Test.sqlite")
+  result <- as.list(dbListTables(con))
+  dbDisconnect(con)
+  expect_identical(identical(portal,result), TRUE)
```
Comment (ethanwhite) [inline tests/testthat/test-retriever.R]: Could this be the somewhat simpler:

```
expect_identical(portal, result)
```

File: tests/testthat/test-retriever.R [hunk 5]
```diff
@@ -1,7 +1,107 @@
...
+  rdataretriever::install('portal', "postgres")
+  con <- dbConnect(dbDriver("PostgreSQL"),
+                   user = 'postgres',
+                   host = 'localhost',
+                   password = "",
+                   port = 5432,
+                   dbname = 'testdb')
+  result <- dbGetQuery(con, "SELECT table_name FROM information_schema.tables WHERE table_schema='testschema'")
+  dbDisconnect(con)
+  expect_identical(all(result$table_name %in%  portal), TRUE)
```
Comment (ethanwhite) [inline tests/testthat/test-retriever.R]: Could this be:

```
expect_identical(result$table_name, portal)
```

or it there are potential order differences

```
expect_setequal(result$table_name, portal)
```

Comment (ethanwhite) [reply tests/testthat/test-retriever.R]: Same below.

File: .travis.yml [hunk 6]
```diff
@@ -1,8 +1,61 @@
...
+  # install testthat using terminal
+  - sudo apt-get install r-cran-testthat
+  - sudo Rscript -e 'library(testthat)'
+
+  # R session information
+  - sudo Rscript -e 'sessionInfo()'
+
+script:
+  - source ~/virtualenv/python3.5/bin/activate
+  - which python
```
[the remark this answers is not in the document]
Comment (henrykironde) [reply .travis.yml]: Not really, I intend to leave it there. Incase we fail, we can confirm the paths used.

**Comment by henrykironde** [comment]
note: expect_identical(identical(file.info(file_path)$size, integer(0)), FALSE),
we are using integer because it looked like the functioned assumed it otherwise

**Comment by henrykironde** [comment]
Looks like the tests are not actually running. They cannot see the retriever path. 
I guess i have to include the path to the packages installed.

**Pull request closed without merging**